streams/src/test/java/org/apache/kafka/streams/integration/StreamsUpgradeTestIntegrationTest.java

```diff
@@ -87,9 +88,9 @@ public void testVersionProbingUpgrade() throws InterruptedException {
         final AtomicInteger usedVersion6 = new AtomicInteger();
         final KafkaStreams kafkaStreams6 = buildFutureStreams(usedVersion6);
         startSync(kafkaStreams6);
-        assertThat(usedVersion6.get(), is(LATEST_SUPPORTED_VERSION + 1));
-        assertThat(usedVersion5.get(), is(LATEST_SUPPORTED_VERSION + 1));
-        assertThat(usedVersion4.get(), is(LATEST_SUPPORTED_VERSION + 1));
+        retryOnExceptionWithTimeout(() -> assertThat(usedVersion6.get(), is(LATEST_SUPPORTED_VERSION + 1)));
```

Comment:
Should we consider doing the same for second / first roll as well?

Reply:
Thanks, @guozhangwang . It _should_ be unnecessary, since they already block until running (in startSync), and they should have gotten the expected number deterministically before transitioning to running. Time will tell if I am wrong, though...